py_ecc/bls/ciphersuites.py

```diff
@@ -123,72 +134,76 @@ def Verify(cls, PK: BLSPubkey, message: bytes, signature: BLSSignature) -> bool:
         return cls._CoreSign(SK, PK + message, cls.DST)
 
     @classmethod
     def Verify(cls, PK: BLSPubkey, message: bytes, signature: BLSSignature) -> bool:
         return cls._CoreVerify(PK, PK + message, signature, cls.DST)
 
     @classmethod
-    def AggregateVerify(cls, pairs: Sequence[Tuple[BLSPubkey, bytes]],
-                        signature: BLSSignature) -> bool:
-        pairs = list(pairs)
-        pairs = [(pk, pk + msg) for pk, msg in pairs]
-        return cls._CoreAggregateVerify(pairs, signature, cls.DST)
+    def AggregateVerify(cls, PKs: Sequence[BLSPubkey],
+                        messages: Sequence[bytes], signature: BLSSignature) -> bool:
+        messages = [pk + msg for pk, msg in zip(PKs, messages)]
+        return cls._CoreAggregateVerify(PKs, messages, signature, cls.DST)
 
 
 class G2ProofOfPossession(BaseG2Ciphersuite):
-    DST = b'BLS_SIG_BLS12381G2-SHA256-SSWU-RO-_POP_'
-    POP_TAG = b'BLS_POP_BLS12381G2-SHA256-SSWU-RO-_POP_'
+    DST = b'BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_'
+    POP_TAG = b'BLS_POP_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_'
 
     @classmethod
-    def AggregateVerify(cls, pairs: Sequence[Tuple[BLSPubkey, bytes]],
-                        signature: BLSSignature) -> bool:
-        return cls._CoreAggregateVerify(pairs, signature, cls.DST)
+    def AggregateVerify(cls, PKs: Sequence[BLSPubkey],
+                        messages: Sequence[bytes], signature: BLSSignature) -> bool:
+        return cls._CoreAggregateVerify(PKs, messages, signature, cls.DST)
 
     @classmethod
     def PopProve(cls, SK: int) -> BLSSignature:
-        pubkey = cls.PrivToPub(SK)
+        pubkey = cls.SkToPk(SK)
         return cls._CoreSign(SK, pubkey, cls.POP_TAG)
 
     @classmethod
     def PopVerify(cls, PK: BLSPubkey, proof: BLSSignature) -> bool:
         return cls._CoreVerify(PK, PK, proof, cls.POP_TAG)
 
     @staticmethod
-    def _AggregatePKs(PKs: Sequence[BLSPubkey]) -> BLSPubkey:
-        accumulator = Z1  # Seed with the point at infinity
-        for pk in PKs:
-            pubkey_point = pubkey_to_G1(pk)
-            accumulator = add(accumulator, pubkey_point)
-        return G1_to_pubkey(accumulator)
+    def _AggregatePKs(PKs: Sequence[BLSPubkey]) -> Union[BLSPubkey, bool]:
+        try:
+            if len(PKs) < 1:
+                raise ValidationError('Insufficient number of PKs. (n < 1)')
+            accumulator = Z1  # Seed with the point at infinity
+            for pk in PKs:
+                pubkey_point = pubkey_to_G1(pk)
+                accumulator = add(accumulator, pubkey_point)
+            return G1_to_pubkey(accumulator)
+        except ValidationError:
+            return False
 
     @classmethod
     def FastAggregateVerify(cls, PKs: Sequence[BLSPubkey],
                             message: bytes, signature: BLSSignature) -> bool:
         aggregate_pubkey = cls._AggregatePKs(PKs)
+        if isinstance(aggregate_pubkey, bool):
+            return False
```

Comment:
If we throw an exception in `_AggregatePKs`, here we can catch it because it's a verification function.
```suggestion
        try:
            aggregate_pubkey = cls._AggregatePKs(PKs)
        except Exception:
            return False
```